pandas/core/groupby.py

```diff
@@ -1691,6 +1691,53 @@ def tail(self, n=5):
         mask = self._cumcount_array(ascending=False) < n
         return self._selected_obj[mask]
 
+    def pipe(self, func, *args, **kwargs):
+        """ Apply a function with arguments to this GroupBy object
+
+        .. versionadded:: 0.21.0
+
+        Parameters
+        ----------
+        func : callable or tuple of (callable, string)
+            Function to apply to this GroupBy object or, alternatively, a
+            ``(callable, data_keyword)`` tuple where ``data_keyword`` is a
+            string indicating the keyword of ``callable`` that expects the
+            GroupBy object.
+        args : iterable, optional
+               positional arguments passed into ``func``.
+        kwargs : dict, optional
+                 a dictionary of keyword arguments passed into ``func``.
+
+        Returns
+        -------
+        object : the return type of ``func``.
+
+        Notes
+        -----
+        Use ``.pipe`` when chaining together functions that expect
+        Series, DataFrames or GroupBy objects. Instead of writing
+
+        >>> f(g(h(df.groupby('group')), arg1=a), arg2=b, arg3=c)
+
+        You can write
+
+        >>> (df
+        ...    .groupby('group')
+        ...    .pipe(f, arg1)
+        ...    .pipe(g, arg2)
+        ...    .pipe(h, arg3))
+
+        See more :ref:`here
+        <http://pandas.pydata.org/pandas-docs/stable/groupby.html#pipe>`
+
+        See Also
+        --------
+        pandas.Series.pipe
+        pandas.DataFrame.pipe
+        pandas.GroupBy.apply
```

Comment:
Can you add a short note on the difference here? (and also add a See also in apply to pipe, again with a short note on the difference)

Reply:
I tried, but admittedly found it very hard to do in so few lines, see result. Do you have suggestions?

Reply:
Something like: "Apply function to each group instead of to the full GroupBy object" ?

Reply:
(IIUC) perhaps

    ``apply`` applies a function to each group. ``pipe`` applies a function to a ``GroupBy`` object.

would work

Reply:
Ok, I've uploaded new ones.

I've built the documentation, and pipe doesn't show up in the API. Do I need to add something somewhere?

Reply:
Also, In groupby.rst line 1178 I have a link ":ref:`here <basics.pipe>`". 

This link doesn't show up in the docs, can anyone see why? It seems normal